sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectExecutionManager.scala

```diff
@@ -61,6 +60,7 @@ private[connect] class SparkConnectExecutionManager() extends Logging {
   private var lastExecutionTimeMs: Option[Long] = Some(System.currentTimeMillis())
```

Comment:
@juliuszsompolski While reviewing my own change, I noticed that lastExecutionTimeMs is never used in the code. Is this dead code? If so, can I just remove it in this PR?

Reply:
`lastExecutionTimeMs` was originally introduced so that some system managing the Spark cluster could check if it's idle and for how long it's idle. Such a monitoring system would call SparkConnectExecutionManager.listActiveExecutions to either know what queries are running in the system, or otherwise learn since when the system is idle.
I am not sure if it got to be used in practice.

Reply:
Thanks, I think I'll have to keep it, and need to deal with a potential data race between "lastExecutionTimeMs" and "executions" => maybe, by not setting None to it and making sure that it monotonically increases? I'll think about it.

Reply:
I think it's fine, because before setting it None, under lock, you check again whether the map is empty. So if some new execution was added concurrently, this will be fine. And on the side that adds that execution, it first puts it into the map (so the map becomes non-empty) and then sets it to non-None under lock.

So if we enter this block setting it to None after some concurrent add already added something to the map, we're fine because map will be non empty and we won't set it to None.
If the map was in fact empty, that means that any concurrent add didn't yet put the execution in the map. Then that add will set it back to non-None right after putting the execution in the map.

Reply:
Yes, it seems safe, couldn't find a counter example. -> anyway I need to add a check (!empty) before setting it to None.